src/plugins/region_map/public/components/map_choice_options.tsx

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import './map_choice_options.scss';
+import React, { useCallback, useMemo, useState } from 'react';
+import {
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+  EuiCheckableCard,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiTextColor,
+} from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { FormattedMessage } from '@osd/i18n/react';
+import { VisOptionsProps } from 'src/plugins/vis_default_editor/public';
+import { FileLayerField, VectorLayer, IServiceSettings } from '../../../maps_legacy/public';
+import { SelectOption, SwitchOption } from '../../../charts/public';
+import { RegionMapVisParams } from '../../../maps_legacy/public';
+import { DEFAULT_MAP_CHOICE, CUSTOM_MAP_CHOICE } from '../../common';
+
+const mapLayerForOption = ({ layerId, name }: VectorLayer) => ({
+  text: name,
+  value: layerId,
+});
+
+const mapFieldForOption = ({ description, name }: FileLayerField) => ({
+  text: description,
+  value: name,
+});
+
+const mapCustomJoinFieldForOption = ({ description, name }: FileLayerField) => ({
+  label: name,
+  text: description,
+  value: name,
+});
+
+export type MapChoiceOptionsProps = {
+  getServiceSettings: () => Promise<IServiceSettings>;
+} & VisOptionsProps<RegionMapVisParams>;
+
+function MapChoiceOptions(props: MapChoiceOptionsProps) {
+  const { getServiceSettings, stateParams, vis, setValue } = props;
+  const vectorLayers = vis.type.editorConfig.collections.vectorLayers;
+  const customVectorLayers = vis.type.editorConfig.collections.customVectorLayers;
+  const vectorLayerOptions = useMemo(() => vectorLayers.map(mapLayerForOption), [vectorLayers]);
+  const customVectorLayerOptions = useMemo(() => customVectorLayers.map(mapLayerForOption), [
+    customVectorLayers,
+  ]);
+  const [userSelectedLayer, setUserSelectedLayer] = useState(DEFAULT_MAP_CHOICE);
+
+  const fieldOptions = useMemo(
+    () =>
+      ((stateParams.selectedLayer && stateParams.selectedLayer.fields) || []).map(
+        mapFieldForOption
+      ),
+    [stateParams.selectedLayer]
+  );
+
+  const customFieldOptions = useMemo(
+    () =>
+      ((stateParams.selectedCustomLayer && stateParams.selectedCustomLayer.fields) || []).map(
+        mapCustomJoinFieldForOption
+      ),
+    [stateParams.selectedCustomLayer]
+  );
+
+  const selectDefaultVectorMap = () => {
+    setUserSelectedLayer(DEFAULT_MAP_CHOICE);
+    setValue('layerChosenByUser', DEFAULT_MAP_CHOICE);
+  };
+
+  const selectCustomVectorMap = () => {
+    setUserSelectedLayer(CUSTOM_MAP_CHOICE);
+    setValue('layerChosenByUser', CUSTOM_MAP_CHOICE);
```

Comment:
Why do we need to duplicate the state? don't you already have access to it in stateParams?